realm/realm-library/src/androidTestObjectServer/java/io/realm/SessionTests.java

```diff
@@ -334,4 +335,30 @@ public void onError(SyncSession session, ObjectServerError error) {
         // Trigger error
         SyncManager.simulateClientReset(SyncManager.getSession(config));
     }
+
+    @Test
+    @UiThreadTest
+    public void uploadAllLocalChanges_throwsOnUiThread() throws InterruptedException {
+        SyncUser user = createTestUser();
+        Realm realm = Realm.getInstance(configuration);
+        try {
+            SyncManager.getSession(configuration).uploadAllLocalChanges();
```

Comment:
Assertion seems to be missing?

Reply:
which assertion? this test didn't change, but now you mention it I've noticed that we don't need the `createTestUser` in this test (same for `downloadAllServerChanges_throwsOnUiThread`)

Reply:
I mean that you don't seem to `fail()` the test if the exception is not thrown.

Reply:
good catch 👍🏼 I thought we were using `@Test(expected = IllegalStateException.class)`